x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvContainsTests.java

```diff
@@ -0,0 +1,368 @@
+                DataType.BOOLEAN,
+                equalTo(result)
+            );
+        }));
+
+        suppliers.add(new TestCaseSupplier(List.of(DataType.CARTESIAN_POINT, DataType.CARTESIAN_POINT), () -> {
+            List<Object> field1 = randomList(1, 10, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomPoint())));
+            List<Object> field2 = randomList(1, 10, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomPoint())));
+            var result = field1.containsAll(field2);
+            return new TestCaseSupplier.TestCase(
+                List.of(
+                    new TestCaseSupplier.TypedData(field1, DataType.CARTESIAN_POINT, "field1"),
+                    new TestCaseSupplier.TypedData(field2, DataType.CARTESIAN_POINT, "field2")
+                ),
+                "MvContainsBytesRefEvaluator[field1=Attribute[channel=0], field2=Attribute[channel=1]]",
+                DataType.BOOLEAN,
+                equalTo(result)
+            );
+        }));
+
+        suppliers.add(new TestCaseSupplier(List.of(DataType.GEO_SHAPE, DataType.GEO_SHAPE), () -> {
+            var field1 = randomList(1, 3, () -> new BytesRef(GEO.asWkt(GeometryTestUtils.randomGeometry(randomBoolean(), 500))));
+            var field2 = randomList(1, 3, () -> new BytesRef(GEO.asWkt(GeometryTestUtils.randomGeometry(randomBoolean(), 500))));
+            var result = field1.containsAll(field2);
+            return new TestCaseSupplier.TestCase(
+                List.of(
+                    new TestCaseSupplier.TypedData(field1, DataType.GEO_SHAPE, "field1"),
+                    new TestCaseSupplier.TypedData(field2, DataType.GEO_SHAPE, "field2")
+                ),
+                "MvContainsBytesRefEvaluator[field1=Attribute[channel=0], field2=Attribute[channel=1]]",
+                DataType.BOOLEAN,
+                equalTo(result)
+            );
+        }));
+
+        suppliers.add(new TestCaseSupplier(List.of(DataType.CARTESIAN_SHAPE, DataType.CARTESIAN_SHAPE), () -> {
+            var field1 = randomList(1, 3, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean(), 500))));
+            var field2 = randomList(1, 3, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean(), 500))));
+            var result = field1.containsAll(field2);
+            return new TestCaseSupplier.TestCase(
+                List.of(
+                    new TestCaseSupplier.TypedData(field1, DataType.CARTESIAN_SHAPE, "field1"),
+                    new TestCaseSupplier.TypedData(field2, DataType.CARTESIAN_SHAPE, "field2")
+                ),
+                "MvContainsBytesRefEvaluator[field1=Attribute[channel=0], field2=Attribute[channel=1]]",
+                DataType.BOOLEAN,
+                equalTo(result)
+            );
+        }));
+    }
+
+    // Adjusted from static method anyNullIsNull in {@code AbstractScalarFunctionTestCase#}
```

Comment:
Can you add a line describing exactly what changed here? So it's easier later to remove this override if/when we extend the original to handle this case

Reply:
Added in https://github.com/elastic/elasticsearch/pull/133636/commits/23d92f0823b51d704875de9208bda532dc0335b7 but want to leave this unresolved as well to follow up - need to look into making the test class more adaptive.